jar-connector/src/main/java/com/sixsq/slipstream/connector/ConnectorBase.java

```diff
@@ -294,18 +294,31 @@ protected String getPrivateSshKeyFileName() throws ConfigurationException, Valid
         return privateSshKeyFile;
     }
 
+    public static String getServerPublicSshKeyFilename() throws ConfigurationException, ValidationException {
+		return Configuration.getInstance().getProperty(ServiceConfiguration.CLOUD_CONNECTOR_ORCHESTRATOR_PUBLICSSHKEY);
+    }
+
     private String getUserPublicSshKey(User user) throws ValidationException, IOException {
 		return user.getParameter(
 				ExecutionControlUserParametersFactory.CATEGORY + "." + UserParametersFactoryBase.SSHKEY_PARAMETER_NAME)
 				.getValue();
     }
 
+    private String getUserOrServerPublicSshKey(User user) throws ValidationException, IOException {
```

Comment:
Providing something that is a valid key will have the same concerns as providing the server's public key--it looks like a backdoor has been added.  I think that we can leave it like this for now, but we should add a task to investigate how to force all of the cloud connectors to not configure SSH for the user in the case that they've not provided their own key.